### PR TITLE
FR - CMS chapter

### PR DESCRIPTION
**Pull request by JustinyAhin**
Makes progress on #539

I've finally ended up the chapter. It was difficult with  many of technical expressions not having an easy or obvious translation in FR. I'd be happy to get a quick review on the PR.

### CODE REVIEW

File: src/content/fr/2019/cms.md [hunk 1]
```diff
@@ -0,0 +1,662 @@
+---
+part_number: III
+chapter_number: 14
+title: CMS
+description: Chapitre CMS de l'Almanach Web 2019 couvrant l'adoption des CMS, la façon dont les solutions CMS sont construites, l'expérience utilisateur des sites web propulsés par les CMS et l'innovation des CMS.
+authors: [ernee, amedina]
+reviewers: []
```
Comment (bazzadp) [inline src/content/fr/2019/cms.md]: Leave this as the English reviewer (not the translator reviewer):

```suggestion
reviewers: [sirjonathan]
```

File: src/content/fr/2019/cms.md [hunk 2]
```diff
@@ -0,0 +1,662 @@
...
+translators: [JustinyAhin]
+discuss:
```
Comment (bazzadp) [inline src/content/fr/2019/cms.md]: Don't think we plan on having language specific discussion forums at this time, so again just leave as English one:

```suggestion
discuss: 1769
```

File: src/content/fr/2019/cms.md [hunk 3]
```diff
@@ -0,0 +1,662 @@
+---
+part_number: III
+chapter_number: 14
+title: CMS
+description: Chapitre CMS de l'Almanach Web 2019 couvrant l'adoption des CMS, la façon dont les solutions CMS sont construites, l'expérience utilisateur des sites web propulsés par les CMS et l'innovation des CMS.
+authors: [ernee, amedina]
+reviewers: []
+translators: [JustinyAhin]
+discuss: 
+published:
```
Comment (bazzadp) [inline src/content/fr/2019/cms.md]: Leave as English date for now - will do a bulk update when French version goes live, but current timestamp will remind us of the format

```suggestion
published: 2019-11-11T00:00:00.000Z
```

File: src/content/fr/2019/cms.md [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+part_number: III
+chapter_number: 14
+title: CMS
+description: Chapitre CMS de l'Almanach Web 2019 couvrant l'adoption des CMS, la façon dont les solutions CMS sont construites, l'expérience utilisateur des sites web propulsés par les CMS et l'innovation des CMS.
+authors: [ernee, amedina]
+reviewers: []
+translators: [JustinyAhin]
+discuss: 
+published: 
+last_updated:
```
Comment (bazzadp) [inline src/content/fr/2019/cms.md]: Leave as English date for now - will do a bulk update when French version goes live, but current timestamp will remind us of the format

```suggestion
last_updated: 2019-11-23T00:00:00.000Z
```

File: src/content/fr/2019/cms.md [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+reviewers: []
+translators: [JustinyAhin]
+discuss: 
+published: 
+last_updated: 
+---
+
+## Introduction
+
+Le terme général **Système de gestion de contenu (SGC)** désigne les systèmes permettant aux personnes et aux organisations de créer, de gérer et de publier du contenu. Un CMS pour le contenu web, plus précisément, est un système visant à créer, gérer et publier du contenu à consommer et à expérimenter via le web ouvert.
```
[the remark this answers is not in the document]
Comment (borisschapira) [reply src/content/fr/2019/cms.md]: Explicitions "CMS" vu qu'il est utilisé après.

File: src/content/fr/2019/cms.md [hunk 6]
```diff
@@ -0,0 +1,662 @@
...
+
+Chaque CMS met en œuvre un sous-ensemble d'un large éventail de capacités de gestion de contenu et les mécanismes correspondants pour permettre aux utilisateurs de construire facilement et efficacement des sites web autour de leur contenu. Ce contenu est souvent stocké dans un certain type de base de données, ce qui permet aux utilisateurs de le réutiliser partout où cela est nécessaire pour leur stratégie de contenu. Les CMS offrent également des capacités d'administration visant à faciliter le téléversement et la gestion du contenu par les utilisateurs, selon leurs besoins.
+
+Le type et la portée du support fourni par les CMS pour la construction de sites varient considérablement; certains fournissent des modèles prêts à l'emploi qui sont "hydratés" par le contenu de l'utilisateur, et d'autres exigent une participation beaucoup plus importante de l'utilisateur pour la conception et la construction de la structure du site.
```
Comment (borisschapira) [inline src/content/fr/2019/cms.md]:
```suggestion
Le type et la portée du support fourni par les CMS pour la construction de sites varient considérablement ; certains fournissent des modèles prêts à l'emploi qui sont "hydratés" par le contenu de l'utilisateur, et d'autres exigent une participation beaucoup plus importante de l'utilisateur à la conception et à la construction de la structure du site.
```

File: src/content/fr/2019/cms.md [hunk 7]
```diff
@@ -0,0 +1,662 @@
...
+
+Dans ce chapitre, nous cherchons à explorer la surface du vaste univers des CMS, en essayant d'éclairer notre compréhension collective du statu quo des ses écosystèmes et du rôle qu'ils jouent dans la perception des utilisateurs sur la façon dont le contenu peut être consommé et expérimenté sur le web. Notre but n'est pas de fournir une vue exhaustive du paysage des CMS ; nous allons plutôt discuter de quelques aspects liés au paysage des CMS en général, et des caractéristiques des pages web générées par ces systèmes. Cette première édition du Web Almanac établit une base de référence, et à l'avenir nous aurons l'avantage de pouvoir comparer les données avec cette version pour l'analyse des tendances.
```
Comment (borisschapira) [inline src/content/fr/2019/cms.md]:
```suggestion
Dans ce chapitre, nous cherchons à explorer la surface du vaste univers des CMS, en essayant d'éclairer notre compréhension collective du statu quo de ces écosystèmes et du rôle qu'ils jouent dans la perception des utilisateurs sur la façon dont le contenu peut être consommé et expérimenté sur le web. Notre but n'est pas de fournir une vue exhaustive du paysage des CMS ; nous allons plutôt discuter de quelques aspects liés au paysage des CMS en général, et des caractéristiques des pages web générées par ces systèmes. Cette première édition du Web Almanac établit une base de référence, et à l'avenir nous aurons l'avantage de pouvoir comparer les données avec cette version pour l'analyse des tendances.
```

File: src/content/fr/2019/cms.md [hunk 8]
```diff
@@ -0,0 +1,662 @@
...
+
+## Adoption des CMS
+
+<figure>
+  <div class="big-number">40%</div>
+  <figcaption>Figure 2. Pourcentage de pages web propulsées par un CMS.</figcaption>
+</figure>
+
+Aujourd'hui, nous remarquons que plus de 40% des pages web sont propulsées par une plateforme CMS ; 40.01% pour les mobiles et 39.61% pour les ordinateur de bureau plus précisément.
```
Comment (borisschapira) [inline src/content/fr/2019/cms.md]:
```suggestion
Aujourd'hui, nous remarquons que plus de 40&nbsp;% des pages web sont propulsées par une plateforme CMS ; 40,01&nbsp;% pour les mobiles et 39.61% pour les ordinateurs de bureau plus précisément.
```

Comment (borisschapira) [reply src/content/fr/2019/cms.md]: Typo sur "les ordinateur" + exemple de mise en forme des nombre (virgule séparatrice, espace avant le symbole ou l'unité). Je vous laisse voir si vous souhaitez généraliser à tout le contenu.

File: src/content/fr/2019/cms.md [hunk 9]
```diff
@@ -0,0 +1,662 @@
...
+
+Le [projet WordPress] (https://fr.wordpress.org/about/) définit sa mission en ces termes : "*démocratiser la publication*". Certains de ses principaux objectifs sont de faciliter l'utilisation et de rendre le logiciel libre et disponible à tous pour créer du contenu sur le web. Un autre élément important est la communauté inclusive que le projet promeut. Dans presque toutes les grandes villes du monde, on peut trouver un groupe de personnes qui se réunissent régulièrement pour se rencontrer, partager et coder dans un effort de compréhension et de construction sur la plate-forme WordPress. La participation aux rencontres locales et aux événements annuels, ainsi que la participation aux réseaux basés sur le web sont quelques unes des façons dont les contributeurs, les experts, les entreprises et les passionnés de WordPress participent à sa communauté globale.
```
Comment (borisschapira) [inline src/content/fr/2019/cms.md]:
```suggestion
Le [projet WordPress] (https://fr.wordpress.org/about/) définit sa mission en ces termes : "*démocratiser la publication*". Certains de ses principaux objectifs sont de faciliter l'utilisation et de rendre le logiciel libre et disponible à tous pour créer du contenu sur le web. Un autre élément important est la communauté inclusive que le projet promeut. Dans presque toutes les grandes villes du monde, on peut trouver un groupe de personnes qui se réunissent régulièrement pour se rencontrer, partager et coder dans un effort de compréhension et de construction sur la plate-forme WordPress. La participation aux rencontres locales et aux événements annuels, ainsi que la participation aux réseaux basés sur le web sont quelques-unes des façons dont les contributeurs, les experts, les entreprises et les passionnés de WordPress participent à sa communauté globale.
```

File: src/content/fr/2019/cms.md [hunk 10]
```diff
@@ -0,0 +1,662 @@
...
+  </a>
+  <div id="fig14-description" class="visually-hidden">Diagramme en bâtons illustrant l'adoption de formats d'images sur les pages CMS pour les ordinateurs de bureau et les mobiles. Le JPEG représente près de la moitié de tous les formats d'image, le PNG en représente un tiers, le GIF un cinquième, et les 5% restants sont partagés entre le SVG, l'ICO et le WebP. Les ordinateurs de bureau et les téléphones portables ont une adoption presque équivalente.</div>
+  <figcaption id="fig14-caption">Figure 14. Adoption de formats d'images sur les pages CMS.</figcaption>
+</figure>
+
+Cette segmentation n'est peut-être pas surprenante étant donné les cas d'utilisation courants pour ces types d'images. Les SVG pour les logos et les icônes sont courants, tout comme les JPEG sont omniprésents. Le WebP est encore un format optimisé relativement nouveau avec [adoption croissante des navigateurs] (https://caniuse.com/#search=webp). Il sera intéressant de voir comment cela aura un impact sur son utilisation dans les prochaines années dans le monde des CMS.
```
Comment (borisschapira) [inline src/content/fr/2019/cms.md]: Attention ici un espace entre le "]" et le "(".

```suggestion
Cette segmentation n'est peut-être pas surprenante étant donné les cas d'utilisation courants pour ces types d'images. Les SVG pour les logos et les icônes sont courants, tout comme les JPEG sont omniprésents. Le WebP est encore un format optimisé relativement nouveau avec [adoption croissante des navigateurs](https://caniuse.com/#search=webp). Il sera intéressant de voir comment cela aura un impact sur son utilisation dans les prochaines années dans le monde des CMS. 
```

File: src/content/fr/2019/cms.md [hunk 11]
```diff
@@ -0,0 +1,662 @@
...
+Les pages web propulsées par les CMS sont générées par un large ensemble des plateformes CMS existantes. Le choix de ces plateformes est très vaste, et de nombreux facteurs peuvent être pris en compte lorsqu'on décide d'utiliser l'une ou l'autre, notamment:
+
+* Fonctionnalité de base
+* Création/édition de workflows et expérience
+* Barrière d'entrée
+* Possibilité de personnalisation
+* Communauté
+* Et beaucoup d'autres.
+
+Les jeux de données CrUX et HTTP Archive contiennent des pages web propulsées par un ensemble d'environ 103 plateformes CMS. La plupart de ces plates-formes sont très petites en termes de part de marché relative. Pour les besoins de notre analyse, nous nous concentrerons sur les principales plates-formes CMS en termes de leur impact sur le web, tel que reflété par les données. Pour une analyse complète, [voir la feuille de calcul des résultats de ce chapitre] (https://docs.google.com/spreadsheets/d/1FDYe6QdoY3UtXodE2estTdwMsTG-hHNrOe9wEYLlwAw/edit#gid=0).
```
Comment (borisschapira) [inline src/content/fr/2019/cms.md]:
```suggestion
Les jeux de données CrUX et HTTP Archive contiennent des pages web propulsées par un ensemble d'environ 103 plateformes CMS. La plupart de ces plates-formes sont très petites en termes de part de marché relative. Pour les besoins de notre analyse, nous nous concentrerons sur les principales plates-formes CMS en termes de leur impact sur le web, tel que reflété par les données. Pour une analyse complète, [voir la feuille de calcul des résultats de ce chapitre](https://docs.google.com/spreadsheets/d/1FDYe6QdoY3UtXodE2estTdwMsTG-hHNrOe9wEYLlwAw/edit#gid=0).
```

File: src/content/fr/2019/cms.md (hunk 9 above)
Comment (borisschapira) [inline src/content/fr/2019/cms.md]:
```suggestion
Le [projet WordPress](https://fr.wordpress.org/about/) définit sa mission en ces termes : "*démocratiser la publication*". Certains de ses principaux objectifs sont de faciliter l'utilisation et de rendre le logiciel libre et disponible à tous pour créer du contenu sur le web. Un autre élément important est la communauté inclusive que le projet promeut. Dans presque toutes les grandes villes du monde, on peut trouver un groupe de personnes qui se réunissent régulièrement pour se rencontrer, partager et coder dans un effort de compréhension et de construction sur la plate-forme WordPress. La participation aux rencontres locales et aux événements annuels, ainsi que la participation aux réseaux basés sur le web sont quelques unes des façons dont les contributeurs, les experts, les entreprises et les passionnés de WordPress participent à sa communauté globale.
```

File: src/content/fr/2019/cms.md [hunk 12]
```diff
@@ -0,0 +1,662 @@
...
+
+Avec nos expériences des CMS qui sont saturées de ces ressources, nous devons considérer l'impact que cela a sur les visiteurs du site web sur le front-end - leur expérience est-elle rapide ou lente ? De plus, lorsque l'on compare l'utilisation des ressources mobiles et ordinateur de bureau, la quantité de requêtes et le poids ne montrent que peu de différence. Cela signifie que la même quantité et le même poids de ressources alimentent à la fois les expériences CMS sur mobile et sur ordinateur de bureau. La variation de la vitesse de connexion et de la qualité des appareils mobiles ajoute [une autre couche de complexité] (https://medinathoughts.com/2017/12/03/the-perils-of-mobile-web-performance-part-iii/). Plus tard dans ce chapitre, nous utiliserons les données du CrUX pour évaluer l'expérience des utilisateurs dans l'écosystème des CMS.
```
Comment (borisschapira) [inline src/content/fr/2019/cms.md]:
```suggestion
Avec nos expériences des CMS qui sont saturées de ces ressources, nous devons considérer l'impact que cela a sur les visiteurs du site web sur le front-end - leur expérience est-elle rapide ou lente ? De plus, lorsque l'on compare l'utilisation des ressources mobiles et ordinateur de bureau, la quantité de requêtes et le poids ne montrent que peu de différence. Cela signifie que la même quantité et le même poids de ressources alimentent à la fois les expériences CMS sur mobile et sur ordinateur de bureau. La variation de la vitesse de connexion et de la qualité des appareils mobiles ajoute [une autre couche de complexité](https://medinathoughts.com/2017/12/03/the-perils-of-mobile-web-performance-part-iii/). Plus tard dans ce chapitre, nous utiliserons les données du CrUX pour évaluer l'expérience des utilisateurs dans l'écosystème des CMS.
```

File: src/content/fr/2019/cms.md (hunk 10 above)
Comment (borisschapira) [reply src/content/fr/2019/cms.md]: Je crois que l'erreur est à plusieurs endroits. Je n'ai pas tout corrigé, je vous laisse le faire avec un Ctrl+F sur le fichier.

**Comment by JustinyAhin** [comment]
Thank you @borisschapira for reviewing.
And @bazzadp for the flags you raised. I'll integrate them soon.

File: src/content/fr/2019/cms.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (bazzadp) [reply src/content/fr/2019/cms.md]: For my own curiousity do you not capitalise "anglais" to "Anglais" in French as a proper noun like you would in English?

[the remark this answers is not in the document]
Comment (borisschapira) [reply src/content/fr/2019/cms.md]: No, we don't. It must seem curious to you, but it's the same for us, when we have to learn how to capitalize your way ;)

[the remark this answers is not in the document]
Comment (bazzadp) [reply src/content/fr/2019/cms.md]: Coolio. Thought so but just wanted to check.

**Comment by JustinyAhin** [comment]
> Also is it possible for you to run `npm run generate` from the src directory to generate the HTML version as well? If not familiar with npm then let us know and can do that later.

Hey @bazzadp. I run `npm run generate` and it looks to throw an error. Here's the output

`npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! almanac.httparchive.org@0.0.1 generate: `node ./tools/generate`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the almanac.httparchive.org@0.0.1 generate script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?`

So I run `npm install` and `npm run generate` again but I don't file it generates any new file. Can you look at this please? 

Also, I've accepted your others suggestions.

**Comment by JustinyAhin** [comment]
@borisschapira all the changes and typos corrections are integrated :)

**Comment by bazzadp** [comment]
> So I run `npm install` and `npm run generate` again but I don't file it generates any new file. Can you look at this please?

Interesting. It would be good to get this sorted for you.

Questions from me:

- Did `npm install` run correctly and create stuff in your src/node_modules directory?
- Any output or error messages from `npm install`?
- Any output or error messages from `npm run generate` or did it run cleanly this time?
- Did `npm run generate` look like it was iterating through the chapters?
- What version of node are you running?

**Comment by JustinyAhin** [comment]
> * Did `npm install` run correctly and create stuff in your src/node_modules directory?

Yes, here's `npm install` output:

`added 160 packages from 152 contributors and audited 210 packages in 72.774s
found 2 low severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details`

> * Any output or error messages from `npm install`?

No

> * Any output or error messages from `npm run generate` or did it run cleanly this time?

I've got just warnings, not errors

> almanac.httparchive.org@0.0.1 generate C:\Users\Eubertine Ahinon\Desktop\almanac.httparchive.org\src
> node ./tools/generate

(node:7300) UnhandledPromiseRejectionWarning: TypeError: object null is not iterable (cannot read property Symbol(Symbol.iterator))
    at generate_chapters (C:\Users\Eubertine Ahinon\Desktop\almanac.httparchive.org\src\tools\generate\generate_chapters.js:27:45)
    at async C:\Users\Eubertine Ahinon\Desktop\almanac.httparchive.org\src\tools\generate\index.js:9:3
(node:7300) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:7300) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

> * Did `npm run generate` look like it was iterating through the chapters?

No, just run at once

> * What version of node are you running?

node 12.14.1 and npm 6.13.4

**Comment by bazzadp** [comment]
Ah this is because us selfish Mac users never set this up to work properly for windows.

If you temporarily change line 26 of src\tools\generate_chapter.js to the following:

```
    const re = /content\/(.*)\/(.*)\/(.*).md/;
```
to the following
```
    const re = /content\\(.*)\\(.*)\\(.*).md/;
```

Then it will work.

Can you try that and then add the HTML to this PR (but do not commit the change to generate_chapter.js - as don't want to break it for the rest of us).

Raised #622 to see if we can fix this properly.

**Comment by mikegeyser** [comment]
**guiltily puts hand up as selfish mac user**

Sorry, will fix it!

**Comment by bazzadp** [comment]
@JustinyAhin , @mikegeyser has now fixed this permenately in https://github.com/HTTPArchive/almanac.httparchive.org/commit/8693c9c48576b55498b32b7992d938206f1af573 and I've merged that in, so if you merge in from upstream master it should work for you now.

Thanks @mikegeyser !

**Comment by JustinyAhin** [comment]
hey @bazzadp  just run `npm run generate` successfully. some files as markup.html were regenarated, and I commit them. you can look at the commits messages.

**Comment by bazzadp** [comment]
Looks good and merged to master now. Thanks for translating this chaper and hopefully next one will be easier!

Happy with those extra commits too - sometimes we miss things for chapters in flights (like for example if you hadn't been able to generate the HTML for your translation) and the next generate picks them up. So thanks for that 😊

**Comment by JustinyAhin** [comment]
Cool,
I really loved translating the chapter. Thanks to all for your help. It doesn't look like there is a translation to review in the queue. So I'll jump on another chapter.

**Comment by bazzadp** [comment]
Thanks! Go for it.

**Pull request merged**